misc/toml-parser/src/main/java/internal/parser/TomlParser.java

```diff
@@ -0,0 +1,325 @@
+        STToken token = peek();
+        if (token.kind == SyntaxKind.DOUBLE_CLOSE_BRACKET_TOKEN) {
+            return consume();
+        } else {
+            recover(token, ParserRuleContext.DOUBLE_CLOSE_BRACKET);
+            return parseDoubleCloseBracket();
+        }
+    }
+
+    private STNode parseTable() {
+        startContext(ParserRuleContext.TOML_TABLE);
+        STNode openBracket = parseOpenBracket();
+        STNode identifierToken = parseKey();
+        STNode closedBracket = parseCloseBracket();
+        List<STNode> fields = parseKeyValues();
+        endContext();
+        return STNodeFactory.createTableNode(openBracket, identifierToken, closedBracket,
+                STNodeFactory.createNodeList(fields));
+    }
+
+    private List<STNode> parseKeyValues() {
+        List<STNode> fields = new ArrayList<>();
+        STToken nextNode = peek();
+        while (!(nextNode.kind == OPEN_BRACKET_TOKEN || nextNode.kind == DOUBLE_OPEN_BRACKET_TOKEN ||
+                nextNode.kind == EOF_TOKEN)) {
+            STNode stNode = parseKeyValue();
+            fields.add(stNode);
+            nextNode = peek();
+        }
+        return fields;
+    }
+
+    private STNode parseOpenBracket() {
+        STToken token = peek();
+        if (token.kind == SyntaxKind.OPEN_BRACKET_TOKEN) {
+            return consume();
+        } else {
+            recover(token, ParserRuleContext.OPEN_BRACKET);
+            return parseOpenBracket();
+        }
+    }
+
+    private STNode parseCloseBracket() {
+        STToken token = peek();
+        if (token.kind == SyntaxKind.CLOSE_BRACKET_TOKEN) {
+            return consume();
+        } else {
+            recover(token, ParserRuleContext.CLOSE_BRACKET);
+            return parseCloseBracket();
+        }
+    }
+
+    private STNode parseKeyValue() {
+        startContext(ParserRuleContext.KEY_VALUE_PAIR);
+        STNode identifier = parseKey();
+        STNode equals = parseEquals();
+        STNode value = parseValue();
+        endContext();
+        return STNodeFactory.createKeyValue(identifier, equals, value);
+    }
+
+    private STNode parseKey() {
+        STToken token = peek();
+        if (isKey(token)) {
+            return consume();
+        } else {
+            recover(token, ParserRuleContext.KEY);
+            return parseKey();
+        }
+    }
+
+    public static boolean isKey(STToken token) {
+        switch (token.kind) {
+            case UNQUOTED_KEY_TOKEN:
+            case BOOLEAN:
+            case STRING_LITERAL:
+                return true;
+            default:
+                return isNumberValidKey(token);
+        }
+    }
+
+    public static boolean isNumberValidKey(STToken token) {
+        if (token instanceof STLiteralValueToken) {
+            if (token.kind == SyntaxKind.DEC_INT ||
+                    token.kind == SyntaxKind.FLOAT) {
+                return !(token.text().startsWith("+") || token.text().startsWith("-"));
```

Comment:
A better way to do this is to send two separate tokens for the one with the 'sign' and the one without it, and check for the syntax kind. That would be faster as well, compared to checking the string content..
e.g two tokens like `signedNumericLiteral` and `unsignedNumericLiteral`